fix issues where funcs are defined out of order

CJ-Wright opened the pull request.

dacamo76 commented on symbol_exporter/ast_symbol_extractor.py:
```diff
@@ -193,6 +195,21 @@ def _add_symbol_to_star_imports(self, imported_symbol):
+            volume = v["data"].get("symbols_in_volume")
+            if volume:
+                for bad_func_name in volume & set(stripped_names):
+                    volume.remove(bad_func_name)
```
We might also want to remove symbol from undeclared symbols here.

dacamo76 commented on symbol_exporter/ast_symbol_extractor.py:
```diff
@@ -193,6 +195,21 @@ def _add_symbol_to_star_imports(self, imported_symbol):
         default = dict(type=SymbolType.STAR_IMPORT, data=dict(imports=set()))
         self.symbols.setdefault("*", default)["data"]["imports"].add(imported_symbol)
 
+    def post_process_symbols(self):
```
Why not make symbols a property and let that be the only entry point into the results?

dacamo76 commented on symbol_exporter/ast_symbol_extractor.py:
```diff
@@ -16,6 +16,8 @@ class SymbolType(str, Enum):
     CLASS = "class"
     STAR_IMPORT = "star-import"
 
+    BOUNDING_SURFACE_SYMBOL_TYPES = {MODULE, FUNCTION, CLASS}
```
where is this used?

CJ-Wright commented:
I think I've addressed the issues, I'm happy to revisit if you'd like, but I also want to get this round of the database started so we can see where things land.
